lib/query/query_supporters.rb

```diff
@@ -530,25 +530,55 @@ def self.dupes_on_name_and_email(np_id)
       .map(&:flatten)
   end
 
+  def self.dupes_on_name_and_phone(np_id)
+    dupes_expr(np_id)
+      .and_where(
+        "name IS NOT NULL\
+         name != ''\
+         AND phone_index IS NOT NULL \
+         AND phone_index != ''"
+      )
+      .group_by(
+        "name,\
+         phone_index"
+      )
+      .execute(format: 'csv')[1..-1]
+      .map(&:flatten)
+  end
+
   def self.dupes_on_name_and_phone_and_address(np_id)
     dupes_expr(np_id)
       .and_where(
         "name IS NOT NULL\
-         AND phone IS NOT NULL \
-         AND phone != '' \
+         AND phone_index IS NOT NULL \
+         AND phone_index != '' \
          AND address IS NOT NULL \
          AND address != ''"
       )
       .group_by(
         "name,\
-         substring(phone from '(([0-9]+.*)*[0-9]+)'),\
+         phone_index,\
          btrim(lower(address), ' '),\
          substring(zip_code from '(([0-9]+.*)*[0-9]+)')"
       )
       .execute(format: 'csv')[1..-1]
       .map(&:flatten)
   end
 
+  def self.dupes_on_address(np_id)
+    dupes_expr(np_id)
+      .and_where(
+        "address IS NOT NULL \
+         AND address != ''"
+      )
+      .group_by(
+        "btrim(lower(address), ' '),\
+         substring(zip_code from '(([0-9]+.*)*[0-9]+)')"
+      )
+      .execute(format: 'csv')[1..-1]
+      .map(&:flatten)
+  end
```

Comment:
Is this supposed to be in this PR?

Reply:
I added this because we're gonna need that for CCR as part of that tix. I mentioned it on this comment https://github.com/CommitChange/tix/issues/3553#issuecomment-1005028776. I could do a separate pull request for that.

Reply:
No, I don't think it's a problem to include it, I just wasn't clear as to why it was there.